lib/api/src/entities/memory/view/inner.rs

```diff
@@ -0,0 +1,253 @@
+use std::{mem::MaybeUninit, ops::Range};
+use wasmer_types::Pages;
+
+use crate::{
+    buffer::{MemoryBuffer, RuntimeMemoryBuffer},
+    macros::rt::{gen_rt_ty, match_rt},
+    AsStoreRef, Memory, MemoryAccessError,
+};
+
+/// A WebAssembly `memory` view.
+///
+/// A memory view is used to read and write to the linear memory.
+///
+/// After a memory is grown a view must not be used anymore. Views are
+/// created using the Memory.view() method.
+gen_rt_ty!(MemoryView<'a> @derives Debug, derive_more::From ; @path memory::view);
+
+impl<'a> RuntimeMemoryView<'a> {
```

Comment:
Why having two structs `MemoryView` and `RuntimeMemoryView`? It seems incredibly redundant when we can simply collapse the two into one